### PR TITLE
17630 Clear name state on request action change

### PR DESCRIPTION
**Pull request by leodube-aot**
*Issue #:* /bcgov/entity#17630

*Description of changes:*

- Clear name state when the request action changes
- Clear form state when the displayed component changes
- Update error message when expro name field is left empty

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namerequest license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-680-q7ymjjkf.web.app

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-680-q7ymjjkf.web.app

**Review by chenhongjing — APPROVED** [review]
LGTM 😺

File: src/components/new-request/search.vue [hunk 1]
```diff
@@ -860,6 +862,14 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin) {
     this.setNoCorpNum(false)
   }
 
+  /** Resets fields when returned to the Tabs component */
+  @Watch('getDisplayedComponent')
+  watchDisplayedComponent (displayedComponent: string) {
+    if (displayedComponent === 'Tabs') {
+      this.onBusiness(null)
+    }
+  }
```
Comment (severinbeauvais) [inline src/components/new-request/search.vue:871]: This is switching between "Get a Business Name" and "Manage My Name Request", right?

Check also what the "Return" or "Reset" buttons do (once you're past the second or 3rd page). I think some things aren't reset there either.

**Comment by severinbeauvais** [comment]
Also, you really need to rebase :)

**Review by severinbeauvais — APPROVED** [review]

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://namerequest-dev--pr-680-q7ymjjkf.web.app

**Pull request merged**